src/main/haskell/kore/src/Kore/Builtin/Map.hs

```diff
@@ -508,28 +559,51 @@ unify
         -> PureMLPattern level variable
         -> MaybeT m (expanded, proof)
 
-    unify0
-        (DV_ resultSort (BuiltinDomainMap map1))
-        (DV_ _          (BuiltinDomainMap map2))
-      =
-        Monad.Trans.lift $ unifyConcrete resultSort map1 map2
+    unify0 dv1@(DV_ resultSort (BuiltinDomainMap map1)) =
+        \case
+            dv2@(DV_ _ builtin2) ->
+                case builtin2 of
+                    BuiltinDomainMap map2 ->
+                        Monad.Trans.lift $ unifyConcrete resultSort map1 map2
+                    _ ->
+                        (error . unlines)
+                            [ "Cannot unify a builtin Map domain value:"
+                            , show dv1
+                            , "with:"
+                            , show dv2
+                            , "This should have been a sort error."
+                            ]
+            app@(App_ symbol2 args2)
+                | isSymbolConcat hookTools symbol2 ->
+                    -- Accept the arguments of MAP.concat in either order.
+                    Monad.Trans.lift $ case args2 of
+                        [ DV_ _ (BuiltinDomainMap map2), x@(Var_ _) ] ->
+                            unifyFramed1 resultSort map1 symbol2 map2 x
+                        [ x@(Var_ _), DV_ _ (BuiltinDomainMap map2) ] ->
+                            unifyFramed1 resultSort map1 symbol2 map2 x
+                        [ _, _ ] ->
+                            give symbolOrAliasSorts
+                                (Builtin.unifyUnsolved dv1 app)
+                        _ ->
+                            Builtin.wrongArity "MAP.concat"
+                | isSymbolElement hookTools symbol2 ->
```

Comment:
Hi, it's late in the night and I'm not fully trusting my ability to review code.

My current guess is that this handles the case when we have a map matching without a framing variable, i.e. `<store> X -> Y </store>`, and that this will match a map with a single element.

I also guess that the same case, but with framing variables, i.e. `<store> X -> Y ... </store>`, is handled by the `[ _, _ ]` case branch above.

Is that correct?